docs/codeql/codeql-for-visual-studio-code/working-with-codeql-packs-in-visual-studio-code.rst

```diff
@@ -0,0 +1,30 @@
+:tocdepth: 1
+
+.. _working-with-codeql-packs-in-visual-studio-code:
+
+Working with CodeQL packs in Visual Studio Code
+===============================================
+
+.. include:: ../reusables/beta-note-package-management.rst
+
+You can view CodeQL packs and write and edit queries for them in Visual Studio Code. 
+
+About CodeQL packs
+------------------
+CodeQL packs are used to create, share, depend on, and run CodeQL queries and libraries. You can publish your own CodeQL packs and download packs created by others. For more information, see ":ref:`About CodeQL packs <about-codeql-packs>`".
+
+Creating and editing CodeQL packs in Visual Studio Code
+-------------------------------------------------------
+To create a new CodeQL pack, you will need to use the CodeQL CLI, which you can do within Visual Studio Code or outside of it with the ``codeql pack init`` command. Once you create an empty pack, you can edit the ``qlpack.yml`` file to add dependencies or change the name or version. For more information, see ":ref:`Creating and working with CodeQL packs <creating-and-working-with-codeql-packs>`".
+
+You can create or edit queries in a CodeQL pack in Visual Studio Code as you would with any CodeQL query, using the standard code editing features such as autocomplete suggestions to find objects to use from the pack's dependencies. 
+
+You can then use the CodeQL CLI to publish your pack to share with others. For more information, see ":ref:`Publishing and using CodeQL packs <publishing-and-using-codeql-packs>`".
+
+Viewing CodeQL packs and their dependencies in Visual Studio Code
+-----------------------------------------------------------------
+Whether you have used the CodeQL CLI to download CodeQL packs that someone else has created, or created your own, you can open the ``qlpack.yml`` file in the root of a CodeQL pack directory in Visual Studio Code and view the dependencies section to see what libraries it depends on.
+
+If you want to understand a query in a CodeQL pack better, you can open the query file and view the code, using the IntelliSense code editing features of Visual Studio Code. For example, if you hover over an object from a library depended on by the pack, Visual Studio Code will resolve it so you can see documentation about the object. 
+
+To view the full definition of an object from a library, you can right-click and choose **Go to Definition**. This will take you to the object definition in the library in your package cache, the shared location where downloaded dependencies are stored, which is in your home directory by default.
```

Comment:
We tend not to use "object" -- I suggest something like "element of a query" or "entity".

Library pack dependencies may also be found in the current workspace (we will look there first, before looking in the package cache). So I suggest something like "If the library pack is present within the same VS Code workspace, this will take you to the definition within the workspace. Otherwise it will take you to the definition of the library within your package cache ..."

The description of the package cache itself is very good :)